pf/tests/schemashim_test.go

```diff
@@ -174,18 +317,59 @@ func TestSchemaShimRepresentations(t *testing.T) {
 			autogold.Expect(`{
   "resources": {
     "_": {
-      "object_attribute": {
+      "single_nested_block": {
         "element": {
           "resource": {
             "a1": {
+              "optional": true,
               "type": 4
             }
           }
         },
+        "optional": true,
         "type": 6
       }
     }
   }
+}`),
+		},
+		{
+			"list-nested-block",
+			&pb.Provider{
+				AllResources: []pb.Resource{{
+					ResourceSchema: schema.Schema{
+						Blocks: map[string]schema.Block{
+							"list_nested_block": schema.ListNestedBlock{
+								NestedObject: schema.NestedBlockObject{
+									Attributes: map[string]schema.Attribute{
+										"a1": schema.StringAttribute{
+											Optional: true,
+										},
+									},
+								},
+							},
+						},
+					},
+				}},
+			},
+			// TODO: Why is this different to list-nested-attribute? Should it be?
+			autogold.Expect(`{
+  "resources": {
+    "_": {
+      "list_nested_block": {
+        "element": {
+          "resource": {
+            "a1": {
+              "optional": true,
+              "type": 4
+            }
+          }
+        },
+        "optional": true,
+        "type": 5
```

Comment:
Note this is a list type - this means that the CastToTypeObject https://github.com/pulumi/pulumi-terraform-bridge/blob/48fd41bd16e2f7f933bb9390ab7619d52faabb6d/pkg/tfshim/util/types.go#L36 is wrong for both PF and SDKv2 - list types CAN have a resource elem in PF, like in SDKv2

This is the root cause of https://github.com/pulumi/pulumi-terraform-bridge/issues/2180 fixed in https://github.com/pulumi/pulumi-terraform-bridge/pull/2181